users/models.py

```diff
@@ -212,6 +213,27 @@ def delete(self):
         for child in self:
             child.delete()
 
+    def has_accepted_communication_for_notification(self, notification_type: str):
+        """Filter guardians who have accepted communication of the type.
+        If the given notification type is included in the types that
+        needs an acceptance, the queryset will be filtered to guardians
+        who has True in `has_accepted_communication`.
+
+        Args:
+            notification_type (str, NotificationType constant string literals):
+                notification type.
+
+        Returns:
+            GuardianQuerySet: a filtered queryset if notification type needs acceptance,
+                otherwise returns self.
+
+        """
+        if not (
+            notification_type in notification_types_that_needs_communication_acceptance
+        ):
+            return self
```

Comment:
self or self.all()?

Reply:
You have tests that test this so it should work with `self` already, and as it *is* a queryset then it makes sense. IMHO I'd just leave it as `self` myself, I don't see any need for the change to `self.all()`.